From b7b49a04fc21cdb8191414c97f221436b0177776 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Wed, 13 Jun 2018 17:32:58 +0800
Subject: [PATCH 4254/5242] MLK-18605-1 drm/bridge: sec-dsim: add cleanup when
 detach dsi client

commit  8d4b7096080de6250d05ba2ebd583e251ccc428e from
https://source.codeaurora.org/external/imx/linux-imx.git

When the dsi device is detached, the dsi parameters saved when
the dsi device is attached should be cleaned to avoid to be
misused. And besides, add some sanity check along with this
cleanup.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit a02c30a0ed8acc4a136d2281431fa4b07d66b933)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/bridge/sec-dsim.c |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/bridge/sec-dsim.c b/drivers/gpu/drm/bridge/sec-dsim.c
index 5c8bf95..5e89a67 100644
--- a/drivers/gpu/drm/bridge/sec-dsim.c
+++ b/drivers/gpu/drm/bridge/sec-dsim.c
@@ -326,7 +326,14 @@ static int sec_mipi_dsim_host_detach(struct mipi_dsi_host *host,
 {
 	struct sec_mipi_dsim *dsim = to_sec_mipi_dsim(host);
 
-	dev_dbg(dsim->dev, "Nothing to be done\n");
+	if (WARN_ON(!dsim->next && !dsim->panel))
+		return -ENODEV;
+
+	/* clear the saved dsi parameters */
+	dsim->lanes	 = 0;
+	dsim->channel	 = 0;
+	dsim->format	 = 0;
+	dsim->mode_flags = 0;
 
 	return 0;
 }
-- 
1.7.9.5

